### ISSUE TITLE
Don't retrieve unused data from the database (performance)

### ISSUE THREAD

**Issue by lyricnz** [LukePrior/nbn-upgrade-map#44]
Current code does 

```python
    cur.execute(f"SELECT * FROM gnaf_202302.address_principals WHERE locality_name = '{target_suburb}' AND state = '{target_state}' LIMIT 100000")
```

On my machine this takes 10.4 seconds for a sample suburb (5957 rows).  Fetching just the five fields we use takes 3.5 seconds.

```python
    query = f"""
        SELECT address, locality_name, postcode, latitude, longitude
        FROM gnaf_202302.address_principals
        WHERE locality_name = '{target_suburb}' AND state = '{target_state}'
        LIMIT 100000"""

    cur.execute(query)
    rows = cur.fetchall()
```

**Comment by lyricnz** [comment]
Time is a bit all over the show.  Query caching?

**Comment by lyricnz** [comment]
FWIW, can convert the result-indexes to column names using something like:

```python
    col_names = [col.name for col in cur.description]

    addresses = []
    for row in rows:
        d = dict(zip(col_names, row))
        address = {
            "name": f"{d['address']} {d['locality_name']} {d['postcode']}",
            "location": [float(d['latitude']), float(d['longitude'])]
        }
        addresses.append(address)
```

**Comment by lyricnz** [comment]
Or even tidier, using the NamedTupleCursor:
- add `cursor_factory=NamedTupleCursor` to the psycopg2.connect() method

then
```python
def get_addresses(target_suburb, target_state):
    """Return a list of addresses for the provided suburb+state from the database."""
    query = f"""
        SELECT address, locality_name, postcode, latitude, longitude
        FROM gnaf_202302.address_principals
        WHERE locality_name = '{target_suburb}' AND state = '{target_state}'
        LIMIT 100000"""  # 100000

    cur.execute(query)

    addresses = []
    while row := cur.fetchone():
        address = {
            "name": f"{row.address} {row.locality_name} {row.postcode}",
            "location": [float(row.latitude), float(row.longitude)]
        }
        addresses.append(address)

    return addresses
```

**Comment by lyricnz** [comment]
Here's a PR in my fork that implements above https://github.com/lyricnz/nbn-upgrade-map/pull/1

It's currently relative to the "style" PR in https://github.com/LukePrior/nbn-upgrade-map/pull/42

**Comment by LukePrior** [comment]
Awesome, I've just merged #42 and the one on your branch looks good

**Comment by lyricnz** [comment]
@LukePrior here's a PR into your repo for the DB stuff https://github.com/LukePrior/nbn-upgrade-map/pull/46
